apollo/agent/agent.py

```diff
@@ -569,13 +573,16 @@ def _execute(
         }
         context[CONTEXT_VAR_UTILS] = OperationUtils(context)
```

Comment:
Here we're adding utils to the context, but utils is also referencing the context, creating a cyclic reference.
This is fine and the GC will eventually release it, but it causes the client to be released later than what we need.